cape/cape_result.py

```diff
@@ -2194,16 +2199,55 @@ def process_buffers(
     buffer_res.set_column_order(["Process", "Source", "Buffer"])
     buffer_ioc_table = ResultTableSection("Buffer IOCs")
     buffer_body = []
-
+    buffers = []
     for process, process_details in process_map.items():
         count_per_source_per_process = 0
         process_name_to_be_displayed = f"{process_details.get('name', 'None')} ({process})"
         for call in process_details.get("decrypted_buffers", []):
             buffer = ""
-            if call.get("CryptDecrypt"):
-                buffer = call["CryptDecrypt"]["buffer"]
-            elif call.get("OutputDebugStringA"):
-                buffer = call["OutputDebugStringA"]["string"]
+            iterable = iter(CRYPT_BUFFER_CALLS)
```

Comment:
I was thinking something simpler that avoids possible infinite while loop issues:
```
crypt_api = next((call.get(item) for item in CRYPT_APIS], None)
if crypt_api:
   buffer = call[item]["buffer"]
   # Check if PE in buffer here!
   # TODO
   hash = sha256(buffer.encode()).hexdigest()
   buffers.append((f"{str(process)}-{item}-{hash}", buffer))
else:
    outputdebugstring_api = next((call.get(item) for item in OUTPUTDEBUGSTRING_APIS], None)
    if outputdebugstring_api :
        buffer = call[item]["string"]
        # Check if PE in buffer here!
        # TODO
        hash = sha256(buffer.encode()).hexdigest()
        buffers.append((f"{str(process)}-{item}-{hash}", buffer))
```

Reply:
Tried crypt_api = next((call.get(item) for item in CRYPT_APIS), None), however item in buffer is not defined. This is the problem with this method, we don't seem to be able to get out the actual value that we need ? I also don't see any potential issue with infinite loop since we define a sentinel value which is checked to break the loop. Anyhow, not sure how to make it work differently in more pythonic ways, not really used to this syntax.